chain/types/ethtypes/eth_types.go

```diff
@@ -55,34 +56,52 @@ const (
 
 type EthUint64 uint64
 
+var _ encoding.TextUnmarshaler = (*EthUint64)(nil)
+var _ encoding.TextMarshaler = (*EthUint64)(nil)
+
+// UnmarshalText should be able to parse these types of input:
+// 1. a string containing a hex-encoded uint64 starting with 0x
+// 2. a string containing an uint64 in decimal
+func (e *EthUint64) UnmarshalText(text []byte) error {
+	s := string(text)
+	base := 10
+	if strings.HasPrefix(s, "0x") {
+		base = 16
+		s = s[2:]
+	}
+	parsedInt, err := strconv.ParseUint(s, base, 64)
+	if err != nil {
+		return xerrors.Errorf("cannot parse uint64: %w", err)
+	}
+	eint := EthUint64(parsedInt)
+	*e = eint
+	return nil
+}
+
+// MarshalText implements encoding.TextMarshaler.
+func (e *EthUint64) MarshalText() ([]byte, error) {
+	return []byte(e.Hex()), nil
+}
+
+// MarshalJSON implements json.Marshaler.
 func (e EthUint64) MarshalJSON() ([]byte, error) {
 	return json.Marshal(e.Hex())
 }
 
 // UnmarshalJSON should be able to parse these types of input:
 // 1. a JSON string containing a hex-encoded uint64 starting with 0x
 // 2. a JSON string containing an uint64 in decimal
-// 3. a string containing an uint64 in decimal
+// 3. a numeral literal containing an uint64 in decimal
 func (e *EthUint64) UnmarshalJSON(b []byte) error {
-	var s string
-	if err := json.Unmarshal(b, &s); err == nil {
-		base := 10
-		if strings.HasPrefix(s, "0x") {
-			base = 16
-			s = s[2:]
-		}
-		parsedInt, err := strconv.ParseUint(s, base, 64)
-		if err != nil {
-			return err
-		}
-		eint := EthUint64(parsedInt)
-		*e = eint
-		return nil
-	} else if eint, err := strconv.ParseUint(string(b), 10, 64); err == nil {
-		*e = EthUint64(eint)
-		return nil
+	// if string literal, remove quotes
+	if b[0] == '"' && b[len(b)-1] == '"' {
```

Comment:
Use `json.Unmarshal`. This won't handle:

1. Strings < 2 bytes.
2. JSON escape sequences.

Etc.